daemon/src/connection.rs

```diff
@@ -116,9 +116,13 @@ impl State {
             return false;
         }
 
+        let heart_beat = self
+            .last_heartbeat()
+            .map(|heart_beat| format!("{heart_beat}"))
+            .unwrap_or_else(|| "None".to_string());
```

Comment:
Nit: I usually use to_owned for converting from str to String because it captures the semantics better (you are only going from borrowed -> owned). ToString works based on Display which can be called on all sorts of types.